web-app/js/portal/cart/DownloadPanel.js

```diff
@@ -158,18 +158,37 @@ Portal.cart.DownloadPanel = Ext.extend(Ext.Panel, {
         setViewPortTab(TAB_INDEX_SEARCH);
     },
 
+    hasDuplicateWfsDownloadLinks: function(downloadHandlers) {
+
+        var wfsDownloads = [];
+        Ext.each(downloadHandlers, function(handler) {
+            Ext.each(handler.getDownloadOptions(), function(downloadOption) {
+                if (downloadOption.type == "WFS") {
+                    wfsDownloads.push(downloadOption.textKey)
+                }
+            });
+        });
+        return (wfsDownloads.hasDuplicates());
+    },
+
     _loadMenuItemsFromHandlers: function(processedValues, collection) {
```

Comment:
There's quite a lot of logic to do with creating the menu for each collection.  Consider extracting this out in to a separate class?